### ISSUE TITLE
RightCurlyCheck: False positive with SAME and METHOD_DEF token

### ISSUE THREAD

**Issue by MEZk** [checkstyle/checkstyle#3920]
````
$ cat TestClass.java
public class TestClass {
    private void foo19() {int var1 = 5;
        var1 = 6;} // line 3
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
            <module name="RightCurly">
            <property name="shouldStartLine" value="true"/>
             <property name="tokens" value="LITERAL_TRY, LITERAL_CATCH, LITERAL_FINALLY, LITERAL_IF, 
                                       LITERAL_ELSE, CLASS_DEF, METHOD_DEF, CTOR_DEF, LITERAL_FOR, LITERAL_WHILE, LITERAL_DO, 
                                       STATIC_INIT, INSTANCE_INIT"/>
            <property name="option" value="same"/>
    </module>
    </module>
</module>

$ java -jar checkstyle-7.6-all.jar -c TestConfig.xml TestClass.java
Starting audit...
[ERROR] TestClass.java:3:18: '}' at column 18 should have line break before. [RightCurly]
[ERROR] TestClass.java:4:1: '}' at column 1 should be on the same line as the next part of a multi-block statement (one that directly contains multiple blocks: if/else-if/else, do/while or try/catch/finally). [RightCurly]
Audit done.
Checkstyle ends with 2 errors.
````

The first violation message is incorrect. If I put line break before '}' on line 3 
```java
public class TestClass {
    private void foo19() {int var1 = 5;
        var1 = 6; // line 3
    } // line 4
}
```

there will be a violation:
```
[ERROR] TestClass.java:4:5: '}' at column 5 should be on the same line as the next part of a multi-block statement (one that directly contains multiple blocks: if/else-if/else, do/while or try/catch/finally). [RightCurly]
```

**Comment by romani** [comment]
`shouldStartLine` option need to be removed from Check as it is covered by `option=alone_or_singleline`.

 I created new issue for this - #4022

If you agree and do not see any nuance, please close this issue.
